### PR TITLE
scheduler's NodeInfo tracks PodInfos instead of Pods

### PR DESCRIPTION
**Pull request by ahg-g**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Modifies NodeInfo to track lists of framework.PodInfo instead of v1.Pod. 

**Which issue(s) this PR fixes**:
Part of #89528

**Does this PR introduce a user-facing change?**:
```release-note
None
```

/cc @alculquicondor

### CODE REVIEW

**Comment by ahg-g** [comment]
Followup PRs will: 
1) NodeInfo will be cleaned up to remove all accessor methods so that PodInfo and NodeInfo have similar design
2) Update NodeInfo.AddPod to accept a PodInfo instead of a Pod.

File: pkg/scheduler/internal/cache/interface.go [hunk 1]
```diff
@@ -56,7 +57,8 @@ import (
 // - Both "Expired" and "Deleted" are valid end states. In case of some problems, e.g. network issue,
 //   a pod might have changed its state (e.g. added and deleted) without delivering notification to the cache.
 type Cache interface {
-	framework.PodLister
```
Comment (Huang-Wei) [inline pkg/scheduler/internal/cache/interface.go:59]: PodLister (returning Pods) is still used as `SharedLister#Pods()`, I suppose you will follow up with cleaning up that, right?

Comment (ahg-g) [reply pkg/scheduler/internal/cache/interface.go:59]: Yes, likely.

**Comment by alculquicondor** [comment]
looks good. I'll wait for verify to pass

**Comment by ahg-g** [comment]
> looks good. I'll wait for verify to pass

Could've lgtmed, a failed verify would block the merge anyway :)

**Pull request merged**